prow/cmd/deck/pr_history.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+)
+
+const (
+	prefixLength         = 7
+	githubPRTemplate     = "https://github.com/%s/%s/pull/%s"
+	githubCommitTemplate = "https://github.com/%s/%s/commit/%s"
+	jobHistLinkTemplate  = "/job-history/%s/pr-logs/directory/%s"
+	// expectedPRKey indicates the key format we expect given the GCS layout specified here:
+	// https://github.com/kubernetes/test-infra/tree/master/gubernator#gcs-bucket-layout
+	expectedPRKey = "<bucket-name>/pr-logs/pull/<org>_<repo>/<pr-number>"
```

Comment:
`<org>_<repo>` segment here isn't technically accurate -- AFAIR k8s uses the legacy system so if you are in the default org or repo, they're omitted

Reply:
I'll just link to the Gubernator docs here instead of explaining this inline